### PR TITLE
bump rust-polars to 0.32.0

### LINKED ISSUE #324: bump rust-polars 0.31.0
[Rust polars new release ](https://github.com/pola-rs/polars/pull/9898)

When this PR is merged revisit optimization toggles from #323  to add unit tests which will now be possible with [`get_current_optimization()`](https://github.com/pola-rs/polars/pull/9883)

### PR DESCRIPTION
**Pull request by sorhawell**
Close #324

### CODE REVIEW

**Comment by sorhawell** [comment]
new cargo.toml inspired by https://github.com/pola-rs/nodejs-polars/pull/85

File: R/expr__expr.R [hunk 1]
```diff
@@ -3675,16 +3675,18 @@ Expr_reshape = function(dims) {
...
 #' @aliases shuffle
 #' @format NULL
 #' @keywords Expr
 #' @examples
 #' pl$DataFrame(a = 1:3)$select(pl$col("a")$shuffle(seed = 1))
-Expr_shuffle = function(seed = NULL) {
-  seed = seed %||% sample(0:10000, 1L)
-  if (!is.numeric(seed) || any(is.na(seed)) || length(seed) != 1L) pstop(err = "seed must be non NA/NaN numeric scalar")
-  unwrap(.pr$Expr$shuffle(self, seed))
+#' stop("new param + reworked to robj_to - > update tests of shufle")
```
Comment (etiennebacher) [inline R/expr__expr.R:3687]: this shouldn't be in the docs, can you put it as a simple comment with "TODO"? (I didn't review the rest, I just noticed this)

Comment (sorhawell) [reply R/expr__expr.R:3687]: When bumping rust I just fix all the compiler errors at once and not update the R side + docs when changed behavior. This stop() was my personal TODO that prevents merging before removed again. When ready for review they will all be gone I promise ;)

Comment (etiennebacher) [reply R/expr__expr.R:3687]: Yeah I just shouldn't review draft PR ;)

**Comment by etiennebacher** [comment]
FYI Polars 0.32.0 was released today. I don't know how much work it is to bump from 0.30.0 to 0.32.0 but it would be really cool to catch up

**Comment by sorhawell** [comment]
Ha then I don't have to do 31 ;)

**Comment by sorhawell** [comment]
this takes ages alot of breaking changes in rust-polars to fix, but getting there ...

10 unit tests left to fix

**Comment by etiennebacher** [comment]
That's some huge work, thanks! Given the number of breaking changes, I think it would be nice to merge this after 0.8.0 is released. What do you think about this?

If you agree, there are #319 and #361 to merge first (cf your list in #364), are these ready for review?

**Comment by sorhawell** [comment]
@etiennebacher  yeah lets see if we can get them done next week

MSRV test points to rust-polars 32  needs something later than rust 1.65. That might become an issue R-universe and CRAN

**Comment by eitsupi** [comment]
> MSRV test points to rust-polars 32 needs something later than rust 1.65. That might become an issue R-universe and CRAN

I think there is no problem on R-universe because the R-universe's builder is currently installed with the latest Rust.
https://github.com/r-universe-org/base-image/blob/58425379cb893b458cdb5951aeb4945120268a84/Dockerfile#L124-L125

Let's forget about CRAN :)

File: README.Rmd [hunk 2]
```diff
@@ -137,7 +139,7 @@ rust_toolchain_version = brio::read_file("Makefile") |>
 During source installation, some environment variables can be set to enable Rust features and profile changes.
 
 - `RPOLARS_FULL_FEATURES="true"` (Build with nightly feature enabled, requires Rust toolchain `r rust_toolchain_version`)
-- `RPOLARS_PROFILE="release-optimized"` (Build with more optimization, requires Rust 1.66 or later)
+- `RPOLARS_PROFILE="release-optimized"` (Build with more optimization, requires Rust or later)
```
Comment (eitsupi) [inline README.Rmd:142]:
```suggestion
- `RPOLARS_PROFILE="release-optimized"` (Build with more optimization)
```

**Review by eitsupi — APPROVED** [review]
Thanks for working on this!
It seems working fine. Is there any reason why this is still a draft?

**Comment by sorhawell** [comment]
> Thanks for working on this!
> It seems working fine. Is there any reason why this is still a draft?

Oups, now marked for review :)

**Comment by eitsupi** [comment]
I wonder this will also close #324?

**Review by etiennebacher — APPROVED** [review]
I just tweaked the NEWS a bit, that's massive so I'd say let's trust the test suite here 😅 thanks @sorhawell !

File: src/rust/Cargo.toml [hunk 3]
```diff
@@ -35,21 +38,23 @@ opt-level = 3
...
-polars-core = { git = "https://github.com/pola-rs/polars.git", rev = "ee2366b68f35c4b69dfe34cbf1eae107d4ebe97b", default_features = false }
 rayon = "1.6.1"
 serde = { version = "1.0.164", features = [ "derive" ] }
 serde_json = "*"
 smartstring = "1.0.1"
-state = "0.5.3"
+state = "0.6.0"
 thiserror = "1.0.40"
-
+polars-core = {git = "https://github.com/pola-rs/polars.git", rev = "c10be8a6598d2f0b14b9da2c39b61ca1a3dd7af5", default-features = false}
```
Comment (eitsupi) [inline src/rust/Cargo.toml:51]: This commit on the PR pola-rs/polars#10474, not on the main branch.

**Review by eitsupi — APPROVED** [review]
I made some minor updates.
(At first, I installed polars from crates.io, but it didn't work...)

**Pull request merged**